java/src/test/java/gherkin/GenerateAstTest.java

```diff
@@ -0,0 +1,69 @@
+package gherkin;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+
+public class GenerateAstTest {
+    @Test
+    public void parse_all_good_features() throws IOException {
+        String[] args = getAllGoodTestFeatures();
+
+        GenerateAst.main(args);
```

Comment:
I don't see a point in unit testing `GenerateAst` at all. It's already tested with the acceptance tests in `Makefile`. It would make more sense to have unit tests for `Parser`.